### DISCUSSION TITLE
MGCB offers

### DISCUSSION THREAD

**Discussion by besopuzzz** [MonoGame/MonoGame#8164] [Content]
Please you to implement the ability to call custom forms for content parameters. For example, there is a custom SpritemapContentProcessor processor that has a Sprites parameter. After double-clicking on the parameter, a custom form opens in which I can create separate areas for sprites. If I'm not mistaken, this was implemented for the XNA ContentPipeline project, and removed in Monogame.

Also, it would not be bad to let users create their own fields for entering parameters. For example, the Vector2 parameter by entering the text value "5 5" or the Rectangle type "0 0 5 5".

I think that such an extension will give you ample opportunities to configure and download your user assets.

**Comment by ThomasFOG** [comment]
> If I'm not mistaken, this was implemented for the XNA ContentPipeline project, and removed in Monogame.

XNA didn't have such feature, and while not impossible to add, it'd be a huge rework of the tool and...

... not quite what the content pipeline is intended to be. What you are asking for goes beyond "configuration": it's content editing. In that regard both XNA and MonoGame's philosophy is that you'd be using another tool to edit your content.

The current plan for the MGCB GUI, is to streamline adding content to make the GUI as much as possible useless (and if it goes well, retire the GUI entirely).
